src/compiletest/runtest.rs

```diff
@@ -927,6 +927,13 @@ fn check_expected_errors(expected_errors: Vec<errors::ExpectedError>,
         format!("{}:{}:", testfile.display(), ee.line)
     }).collect::<Vec<String>>();
 
+    let (expect_help, expect_note) =
+        expected_errors.iter()
+                        .fold((false, false),
+                              |(acc_help, acc_note), ee|
+                                  (acc_help || ee.kind == "help:", acc_note ||
```

Comment:
so, the logic here is that "if there is a help, then all help must be present", plus "if there is a note, then all notes must be present."

In hindsight I had *thought* that what we were discussing was "if there is either a help or a note, then all help and notes must be present."

I suspect that the logic you have chosen is actually more convenient in practice than what I was expecting, but I still figured I should point it out explicitly, in case someone else was as confused as I was about the proposal.